From c51e6ceb7f1443603e9713d297b0386db91f524e Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 18 Nov 2015 09:03:00 +0000
Subject: [PATCH] Use configure test for struct timespec

Use the configure script to test for struct time spec instead of trying
to keep track of what platforms has it.

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
[Retrieved from:
http://cgit.openembedded.org/meta-openembedded/tree/meta-oe/recipes-support/open-vm-tools/open-vm-tools/0005-Use-configure-test-for-struct-timespec.patch?h=sumo]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[rebased against stable-12.3.0]
Signed-off-by: Stefan Agner <stefan@agner.ch>
[Bernd: rebased against stable-13.0.10]
Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 configure.ac           | 1 +
 lib/include/hgfsUtil.h | 8 +-------
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 89b0bd8db..0ba2f6658 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1393,6 +1393,7 @@ AC_TYPE_OFF_T
 AC_TYPE_PID_T
 AC_TYPE_SIZE_T
 AC_CHECK_MEMBERS([struct stat.st_rdev])
+AC_CHECK_MEMBERS([struct timespec.tv_sec],[],[],[[#include <time.h>]])
 AC_HEADER_TIME
 AC_STRUCT_TM
 AC_C_VOLATILE
diff --git a/lib/include/hgfsUtil.h b/lib/include/hgfsUtil.h
index 609f4c000..a3a022d42 100644
--- a/lib/include/hgfsUtil.h
+++ b/lib/include/hgfsUtil.h
@@ -53,13 +53,7 @@
 #      include <time.h>
 #   endif
 #   include "vm_basic_types.h"
-#   if !defined _STRUCT_TIMESPEC &&   \
-       !defined _TIMESPEC_DECLARED && \
-       !defined __timespec_defined && \
-       !defined sun && \
-       !defined __FreeBSD__ && \
-       !__APPLE__ && \
-       !defined _WIN32
+#   if !defined HAVE_STRUCT_TIMESPEC_TV_SEC
 struct timespec {
    time_t tv_sec;
    long   tv_nsec;
-- 
2.47.3

